(#211) Sass partials and story doms

olitharp-nci opened the pull request:
Closes #211

bryanpizzillo commented:
Some other items that should be considered for this spike PR:
1. Get fonts working, as discussed in slack. Fonts do not work 100% with the shadow dom
2. How are we going to test? The state of the PR now is that we have a way something could be tested, which is good, but we still need the how.
    * The point of this is to find a way to test our partials to make sure that everything is getting pulled in correctly. So does the component **look** and **behave** as it should? Basically if someone modified the base/require let's catch the fact that they broke everything.

olitharp-nci commented:
@bryanpizzillo Font imports need to be in `head` . I moved the sass to the Helmet component that we added to support scripts, font styles should now work as expected.

olitharp-nci commented:
### Shadow dom
- We need a way to make sure we're only bringing in partials we want to test for
- Storybook on default will combine sass to one stylesheet
- Idea was to see if we can use shadow doms and inline styles to utilize style encapsulation
- There is a great package [react-shadow](https://www.npmjs.com/package/react-shadow) that easily sets up shadow dom structure in our react components using `root.div` 

### How to test for scripts?
- Inline `script` is supported with package [react-helment](https://www.npmjs.com/package/react-helmet)

### Problems with font not loading 
- The @font-face CSS at-rule must be declared in the main document, not in the Shadow DOM.
- Importing font faces separately from the rest of the inline sass using `react-helment` fixes this problem
- Fonts are being imported in the `TestCase` component for now

### Visual regression testing
- [Loki](https://github.com/oblador/loki)
- Cypress, cucumber, cypress testing tool
- [BackstopJS](https://github.com/garris/BackstopJS)
- Stoybook addon [storyshots](https://storybook.js.org/docs/react/workflows/snapshot-testing)
- Storybook addon [Pseudo States](https://storybook.js.org/addons/storybook-addon-pseudo-states)

#### Loki
- Great snapshot imagedif 
- Cannot easily override reference images, must explicitly approve changes
- Cannot test pseudo classes or events

#### Cypress
...TODO
